lib/Service/ContactsIntegration.php

```diff
@@ -77,6 +112,16 @@ public function getMatchingRecipient(string $term): array {
 				if ($e === '') {
 					continue;
 				}
+				$lowerTerm = strtolower($term);
+				if ($isSystemUser &&
+					($shareeEnumerationInGroupOnly || $shareeEnumerationFullMatch) &&
+					!($shareeEnumerationInGroupOnly && $isInSameGroup) &&
+					!($shareeEnumerationFullMatch && $lowerTerm !== '' && $lowerTerm === strtolower($fn)) &&
+					!($shareeEnumerationFullMatchUserId && $lowerTerm === strtolower($id)) &&
+					!($shareeEnumerationFullMatchEmail && $lowerTerm === strtolower($e))
+				) {
+					continue;
+				}
```

Comment:
Allow same group and all address books other than the system one as the share api matches should not influence one's own contacts and the limitations of the share API should not apply to them.

```suggestion
if(!$isSystemUser || $isInSameGroup) {
					$receivers[] = [
						'id' => $id,
						'label' => (empty($fn) ? $e : "$fn ($e)"),
						'email' => $e,
						'photo' => $photo,
					];
					continue;
				}

				if($shareeEnumerationFullMatch && $lowerTerm !== '' &&
					((!empty($fn) && strtolower($fn) === $lowerTerm) ||
					($shareeEnumerationFullMatchUserId && strtolower($id) === $lowerTerm )||
					($shareeEnumerationFullMatchEmail && strtolower($e) === $lowerTerm))
				){
					$receivers[] = [
						'id' => $id,
						'label' => (empty($fn) ? $e : "$fn ($e)"),
						'email' => $e,
						'photo' => $photo,
					];
				}
```

Then the function is done.

I have restructured the if condition to be permissive instead of dismissve to be a bit more clear.

Reply:
You right the permissive approach is better ! I propose a compromise though :wink:.

Reply:
hehe I see you like your `if`s short and sweet. Well done!